### ISSUE TITLE
avoid calling costgraph.shortest_paths method when blades are manufactured

### ISSUE THREAD

**Issue by rjhanes** [NREL/celavi#44]
`CostGraph` has a `find_upstream_neighbor` method that will connect a power plant facility with its closes manufacturing facility. When blades are created, a call to `shortest_paths` is used instead to connect the manufacturing and power plant facilities.

This upgrade would swap out the more intensive `shortest_paths` call with a call to `find_upstream_neighbor` when the blade is initialized (already implemented) and use information from that method to generate the first two steps of the blade's queue. 

As a result, the `shortest_paths` method will only be called when a blade reaches end of life - ie once per blade, instead of twice as it currently runs.

**Comment by rjhanes** [comment]
Running the above commits on `tiny-data` at 04d7f63 with the `CostGraph` parameters at bottom produces these results, which look correct:

![blade_count_facets_1632417572](https://user-images.githubusercontent.com/18059389/134554530-435ea6f3-02a3-4987-8321-da99811ecd95.png)

```
costgraph_parameters:
    sc_begin: manufacturing
    sc_end: 
    - landfilling
    - cement co-processing
    - next use
    cg_verbose: 2
    save_cg_csv: True
    finegrind_cumul_initial: 1.0
    finegrind_initial_cost: 5.0
    finegrind_revenue: 60.0
    finegrind_learnrate: -0.05
    finegrind_material_loss: 0.3
    coarsegrind_cumul_initial: 1.0
    coarsegrind_initial_cost: 5.0
    coarsegrind_learnrate: -0.05
    cg_update_timesteps: 12
# coprocessing revenue at default value
```

Next step: test on `medium-data`

**Comment by rjhanes** [comment]
Accumulation in next use facilities is fixed: (same data branch and commit as above, same `CostGraph` parameters as above)

![blade_count_facets_1632417960](https://user-images.githubusercontent.com/18059389/134555212-8449e325-be47-4765-b024-a000004967ba.png)

**Comment by rjhanes** [comment]
Closed with PR #50
